type.go

```diff
@@ -1316,7 +1317,20 @@ func (t *jsonType) Decode(dst encoding.Values, src []byte, enc encoding.Encoding
 }
 
 func (t *jsonType) AssignValue(dst reflect.Value, src Value) error {
-	return ByteArrayType.AssignValue(dst, src)
+	switch dst.Kind() {
+	// Assign value using ByteArrayType for BC...
+	case reflect.String, reflect.Slice:
```

Comment:
It feels like we shouldn't need to dispatch to the `ByteArrayType` at all... but i kept this in - in case I'm wrong.